### DISCUSSION TITLE
can the dumped file is used by ida7.5？

### DISCUSSION THREAD

**Discussion by zbx911** [vfsfitvnm/frida-il2cpp-bridge#113] [Q&A]
dump()  => file.cs => used by ida7.5?

**Comment by vfsfitvnm** [comment]
It may.
Write your own ida script to parse the `file.cs` and do what you need (I did the same for ghidra), or write your own dump function if you need something more specific.

**Comment by zbx911** [comment]
> 
> 
> It may.
> Write your own ida script to parse the `file.cs` and do what you need (I did the same for ghidra), or write your own dump function if you need something more specific.

can you share ghidra script?

**Comment by vfsfitvnm** [comment]
@zbx911 
```py
from ghidra.program.model.address import AddressFactory
from ghidra.program.model.symbol.SourceType import USER_DEFINED
import re

address_factory = getAddressFactory()

with open("/path/to/dump.cs", "r") as file:
    content = file.read()

matches = re.findall("([^\s]+)(?:\(.+)(0x[0123456789abcdef]{8})", content)

for match in matches:
    function = getFunctionAt(address_factory.getAddress(match[1]))
    if function:
        function.setName(match[0], USER_DEFINED)
```
